src/restore-token.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2023 Red Hat
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+
+#include "session.h"
+
+typedef enum _PersistMode
+{
+  PERSIST_MODE_NONE = 0,
+  PERSIST_MODE_TRANSIENT = 1,
+  PERSIST_MODE_PERSISTENT = 2,
+} PersistMode;
+
+void set_transient_permissions (const char *sender,
+                                const char *restore_token,
+                                GVariant *restore_data);
```

Comment:
I've been staring at this for a while, and am starting to this it'd be better to rename these functions and he enum to something like:

```
typedef enum _XdpSessionPersistMode
{
   ...
} XdpSessionPersistMode;

void xdp_session_persistence_store_transient (...);
void xdp_session_persistence_store (...);
GVariant * xdp_session_persistence_find (...);
```

What do you think?

Reply:
I guess you mean taking the `sender` from the session? Seems reasonable to me.

Reply:
I was thinking only in renaming the functions, arguments would stay the same. But now that you mention, taking sender from the session might be a good idea too!

Reply:
Anything called `xdp_session_*()` I'd expect to take a "session as the first argument.

Reply:
From Matrix:

> all of them
> the ones you listed already receive a Session* object; the others (`set_transient_permissions`, `delete_transient_permissions`, `get_transient_permissions`, etc...) can be adjusted to receive a Session* object too
> we always pass `session->sender` to the ones that receive `const char *sender`, for example